### PR TITLE
Add support for minutesUntilShutdown option, added some more logging

### PR DESCRIPTION
**Pull request by nickeidler**
* Adds a new config option `minutesUntilShutdown ` that allows server admins to customize how long it will take for the server to shut down after the last player leaves
* Added some more logging to the plugin itself (mostly for my own benefit while developing haha)
* Added some random stuff to `.gitignore` that VS Code added

### CODE REVIEW

File: src/main/java/com/robomwm/azuresaver/AzureSaver.java [hunk 1]
```diff
@@ -85,49 +86,61 @@ private void copyFile(File file, String name)
...
             }
         }, true, 25);
     }
 
-    @EventHandler
-    public void onJoinServer(LoginEvent event)
-    {
-        getLogger().info("javadocs succ");
-        if (event.isCancelled())
-            event.setCancelReason("ok boomer");
```
Comment (RoboMWM) [inline src/main/java/com/robomwm/azuresaver/AzureSaver.java:107]: dang this was a good meme to keep in for anyone poking around even though I discovered later this event is not called lul

**Review by RoboMWM — APPROVED** [review]
Looks good otherwise. I've left this in favor of AzureResizer (and its standalone counterpart so I don't have to shade in a bazillion libraries that can conflict with other plugins) so I'll just trust that your stuff works lol.

**Comment by nickeidler** [comment]
Haha cool. I'm running a very small server (<5 people) so I can't really resize it any smaller than it already is. Only issue with this plugin is that the player login will time out while the VM boots up. Not sure if that is something that can be fixed on the bungee/waterfall side, or if it is just a hardcoded 30s connection timeout in the client.

**Comment by RoboMWM** [comment]
> Only issue with this plugin is that the player login will time out while the VM boots up. Not sure if that is something that can be fixed on the bungee/waterfall side, or if it is just a hardcoded 30s connection timeout in the client.

Yea, idk if there's any way for bungee to "hold" the connection or etc., else the only other idea I had was to have a small server on the same machine or something spun up to just hold players in a void or something until the VM spins up.

I use azureresizer to resize the same VM so I'm only using 1 VM and don't have to deal with bungee. Otherwise it works the same way; when nobody's on it's at the lowest tier possible (1GB ram which MC can't handle anymore - if a player joins. It can kind of handle it if nobody's on), and scales itself up when a player joins. It takes a little longer iirc and your server appears off while it is "waking up" but it's way easier to maintain for my needs.

**Pull request merged**